providers/structured/provider.go

```diff
@@ -104,6 +132,14 @@ func (p *StructuredOutputLogProvider) RecordEvent(ctx context.Context, eventName
 		Metrics:       metrics,
 	}
 	p.recordEventCalls = append(p.recordEventCalls, callArgs)
+
+	if p.LogProvider != nil {
+		p.LogProvider.RecordEvent(ctx, eventName, metrics)
+	}
 }
 
-func (p *StructuredOutputLogProvider) Wait() {}
+func (p *StructuredOutputLogProvider) Wait() {
+	if p.LogProvider != nil {
+		p.LogProvider.Wait()
+	}
+}
```

Comment:
You don't need to implement this at all; the chaining provider will give you an empty default implementation.